src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -199,20 +209,88 @@ def process(
                 if grant_statements is not None:
                     cc.message(grant_statements)
                     collected_output.append(grant_statements)
+            extension_fns_ddl = "\n".join(collected_output)
+            dest_file_py_file_to_ddl_map[py_file] = extension_fns_ddl
+
+        write_to_sql = kwargs.get(
+            "write_to_sql", True
+        )  # temporary until we have better e2e testing mechanism
+
+        if write_to_sql:
+            dest_file_to_sql_file = create_and_write_to_sql_files(
```

Comment:
Subjective, but that's a strange primitive to me. It does a bit too much? Why is it returning a map? Conceptually, I'd probably do something like:

```py
generated_sql_dir = self.deploy_root / self.generated_dir  # I'm assume generated dir is relative here, but adjust as needed
all_sql_files = []
for py_file, extension_fns in collected_extension_functions_by_path.items():
    sql_file_rel_path = generate_ddl(py_file, extensions_fns, output_dir = self.deploy_root / self.generated_root
    if sql_file_path:  # if DDL generation fails, None is returned
        all_sql_files.append(sql_file_rel_path)
```

if all_sql_files:
    # generate main file here will one execute immediate per file in the list

# weave setup script here